src/terminal/parser/stateMachine.cpp

```diff
@@ -1292,10 +1686,12 @@ void StateMachine::ProcessCharacter(const wchar_t wch)
         _ActionExecute(wch);
         _EnterGround();
     }
-    else if (_isEscape(wch) && _state != VTStates::OscString)
+    else if (_isEscape(wch) && _state != VTStates::OscString && _state != VTStates::DcsPassThrough && _state != VTStates::DcsIntermediate && _state != VTStates::DcsIgnore)

```

Comment:
As discussed above, I think this should only apply to the `DcsPassThrough` state.

Reply:
Seems to me `DcsTermination` is also needed here?

Reply:
My first thought was that you were right, but now that I've had a chance to do some testing, I think that's probably wrong. Consider this test case:

    printf "\ePq\e\e[41mRED\e[m\n"

We've started a DCS sequence, and got into the pass-through state. We receive one escape, which could be the start of an ST, but it's then followed by a second escape. At that point we're clearly aborting the DCS, but what happens next?

Going by the state diagram, I'd expect an escape to start an escape sequence from anywhere, so that second escape is just the start of a CSI which sets the background color to red (and that's what XTerm does). For that to work we can't include `DcsTermination` here.

Reply:
You're right. I can manually confirm your test case works with this PR:

![image](https://user-images.githubusercontent.com/4710575/88250191-5f79aa80-ccd9-11ea-89a5-47261f423103.png)

Reply:
Maybe I'm missing something but I don't see a lot of test cases for complicated escape sequences like these. Is it generally a good practice to write test cases for possibly malformed, garbled or malicious sequences? That's of course not the scope of this PR specifically. Just wondering if it's a possible direction to improve VT tests.